### DISCUSSION TITLE
steam-engine saving

### DISCUSSION THREAD

**Discussion by omeringen** [gildas-lormeau/SingleFile#1482] [Q&A] [answered]
Hi,
I saved http://www.steam-engine.org/coil?a=true but buttons doesn't work on webpage.
Tried to disable _preferences -> network --> scripts_ but no difference

**Comment by gildas-lormeau** [comment]
You have to disable the following options:
 - `HTML content > compress HTML content`
 - `HTML content > remove hidden elements`
 - `HTML content > set content security policy`
 - `Stylesheets > remove unused styles`
 - `Stylesheets > remove stylesheets for alternative devices to screens`
 - `Images > group duplicate images together`
 - `Images > remove images for alternative screen resolutions`
 - `Fonts > remove unused fonts`
 - `Fonts > remove alternative fonts`
 - `Network > blocked resources > scripts`

You have also to enable `HTML content > save raw page`.

Here is in the ZIP file a settings file with these changes applied: [singlefile-settings.json.zip](https://github.com/user-attachments/files/15872636/singlefile-settings.json.zip)

**Reply by omeringen** [reply]
> You have to disable the following options:
> 
>     * `HTML content > compress HTML content`
> 
>     * `HTML content > remove hidden elements`
> 
>     * `HTML content > set content security policy`
> 
>     * `Stylesheets > remove unused styles`
> 
>     * `Stylesheets > remove stylesheets for alternative devices to screens`
> 
>     * `Images > group duplicate images together`
> 
>     * `Images > remove images for alternative screen resolutions`
> 
>     * `Fonts > remove unused fonts`
> 
>     * `Fonts > remove alternative fonts`
> 
>     * `Network > blocked resources > scripts`
> 
> 
> You have also to enable `HTML content > save raw page`.
> 
> Here is in the ZIP file a settings file with these changes applied: [singlefile-settings.json.zip](https://github.com/user-attachments/files/15872636/singlefile-settings.json.zip)

Thanks but doesn't work at all, and it become worse actually. Have you tried saving webpage with these settings ?

**Reply by gildas-lormeau** [reply]
I'm sorry but this a limitation of SingleFile. If you want to save interactive pages, you should save save them in a format like WARC. You can find more info here for example: https://github.com/webrecorder

**Comment by omeringen** [comment, marked as answer]
I handled it successfully with https://github.com/Y2Z/monolith

**Reply by gildas-lormeau** [reply]
Does it work offline?

**Reply by gildas-lormeau** [reply]
Anyway, I cannot reproduce your issue. The page saved with SingleFile seems to work on my end.

**Reply by omeringen** [reply]
> Does it work offline?

yes

**Reply by omeringen** [reply]
> Anyway, I cannot reproduce your issue. The page saved with SingleFile seems to work on my end.

I am on GNU/Linux(arch)+Firefox, might that be the difference ?

**Comment by omeringen** [comment]
I imported your settings file and tried again. Buttons on webpage doesn't work at all.
![2024-07-07_13-53](https://github.com/gildas-lormeau/SingleFile/assets/22679786/0ace225b-f74c-4252-986d-56c235056bba)

**Reply by gildas-lormeau** [reply]
Sorry for that, I will take a loot at the issue.

**Reply by gildas-lormeau** [reply]
I was not able to reproduce your issue with the settings I mentioned. However, I fixed a bug related to the panel at the right. The fix will be available in the next version.

**Reply by gildas-lormeau** [reply]
For the record, here is in the zip file the page saved with the last version of SingleFile. The page is not perfect but functional.

[Coil wrapping _ Steam Engine _ free vaping calculators (23_07_2024 23_24_13).html.zip](https://github.com/user-attachments/files/16354434/Coil.wrapping._.Steam.Engine._.free.vaping.calculators.23_07_2024.23_24_13.html.zip)
